### PR TITLE
BLD: add tf 2.0 dependency for addons

### PR DESCRIPTION
**Pull request by facaiy**
Related with #454

### CODE REVIEW

**Comment by seanpmorgan** [comment]
Thanks @facaiy this is a nice cleanup. Only thing I want to be positive is that `tensorflow-gpu` does not crash on a CPU only box... I believe that is true but haven't tested. Has anyone confirmed this?

We should probably cherry pick and publish 0.5.1 with this as well.

File: setup.py [hunk 1]
```diff
@@ -39,24 +40,44 @@
...
-    project_name = 'tfa-nightly'
-    nightly_idx = sys.argv.index('--nightly')
-    sys.argv.pop(nightly_idx)
-    version['__version__'] += datetime.strftime(datetime.today(), "%Y%m%d")
-else:
-    project_name = 'tensorflow-addons'
+if project_name == TFA_RELEASE:
+    # TODO: remove if-else condition when tf supports package consolidation.
+    if platform.system() == 'Linux':
+        REQUIRED_PACKAGES.append('tensorflow-gpu >= 2.0.0-rc0')
```
Comment (guillaumekln) [inline setup.py]: Is the version requirement `>=` correct? For example if a user froze `tensorflow_addons` to a version built against TensorFlow 2.0 and then update to TensorFlow 2.1, will it work correctly?

**Comment by guillaumekln** [comment]
> Only thing I want to be positive is that tensorflow-gpu does not crash on a CPU only box... I believe that is true but haven't tested. Has anyone confirmed this?

Yes, it works since TensorFlow 1.14.

File: setup.py (hunk 1 above)
Comment (seanpmorgan) [reply setup.py]: Even though backwards compatible API guarantees should be there, there may new features we're using so I think this is a good catch.

Should be `==`

Comment (facaiy) [reply setup.py]: Thanks, done :-)

**Comment by facaiy** [comment]
> We should probably cherry pick and publish 0.5.1 with this as well.

Totally agree, will do later :-)

**Pull request merged**